apps/calendar/test/unit/support/calendar.js

```diff
@@ -0,0 +1,143 @@
+define(function(require, exports) {
+'use strict';
+
+var Db = require('db');
+var Factory = require('test/support/factory');
+var MockProvider = require('test/support/mock_provider');
+var app = require('app');
+var core = require('core');
+
+require('test/support/factories/all');
+
+exports._lastEnvId = 0;
+
+exports.core = function() {
+  var pendingManager = app.pendingManager;
+  if (pendingManager) {
+    // hack to ensure clean tests
+    pendingManager.objects.forEach(item => pendingManager.unregister(item));
+  }
+
+  if (core.db && core.db.isOpen) {
+    core.db.close();
+  }
+
+  app.setupCore('b2g-test-calendar');
+  core.providerFactory.providers.Mock = new MockProvider();
+  return core;
+};
+
+exports.accountEnvironment = function(accOverrides, calOverrides) {
+  accOverrides = accOverrides || {};
+  accOverrides.providerType = accOverrides.providerType || 'Mock';
+  calOverrides = calOverrides || {};
+
+  setup(function(done) {
+    var id = ++exports._lastEnvId;
+    accOverrides._id = id;
+    calOverrides.accountId = id;
+
+    var trans = core.db.transaction(['accounts', 'calendars'], 'readwrite');
+    trans.oncomplete = () => done();
+    trans.onerror = event => done(event.target.error);
+
+    var account = this.account = Factory('account', accOverrides);
+    var calendar = this.calendar = Factory('calendar', calOverrides);
+    var storeFactory = core.storeFactory;
+    var accountStore = storeFactory.get('Account');
+    var calendarStore = storeFactory.get('Calendar');
+    return Promise.all([
+      accountStore.persist(account, trans),
+      calendarStore.persist(calendar, trans)
+    ]);
+  });
+};
+
+exports.eventEnvironment = function(busytimeOverrides, eventOverrides) {
+  eventOverrides = eventOverrides || {};
+  eventOverrides._id = eventOverrides._id || 'one';
+  busytimeOverrides = busytimeOverrides || {};
+
+  setup(function(done) {
+    var storeFactory = core.storeFactory;
+    var busytimeStore = storeFactory.get('Busytime');
+    var eventStore = storeFactory.get('Event');
+    var trans = core.db.transaction(eventStore._dependentStores, 'readwrite');
+    trans.oncomplete = () => done();
+    trans.onerror = event => done(event.target.error);
+
+    eventOverrides.calendarId = this.calendar._id;
+    var event = this.event = Factory('event', eventOverrides);
+    busytimeOverrides.calendarId = this.calendar._id;
+    busytimeOverrides.eventId = this.event._id || eventOverrides._id;
+    var busytime = this.busytime = Factory('busytime', busytimeOverrides);
+    return Promise.all([
+      eventStore.persist(event, trans),
+      busytimeStore.persist(busytime, trans)
+    ]);
```

Comment:
do you really need to return a promise here? shouldn't `trans.oncomplete = () => done();` take care of it?